src/CMakeLists.txt

```diff
@@ -416,7 +416,13 @@ set(ceph_common_deps
   ${CRYPTO_LIBS}
   ${GSSAPI_LIBRARIES}
   ${CMAKE_THREAD_LIBS_INIT}
-  ${CMAKE_DL_LIBS})
+  ${CMAKE_DL_LIBS}
+  ${ZLIB_LIBRARIES})
+if (NOT WIN32)
+  # We have to avoid a circular dependency between the common and
```

Comment:
if this is a circular dependency, we'd better fix on other platforms also.

Reply:
Right, my goal was to keep it simple. I'll look into it again, maybe breaking this circular dependencies wouldn't require too many changes.